### PR TITLE
Remove redundant RayCluster logic from Pod controller

### PR DESCRIPTION
**Pull request by astefanutti**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

A quick follow up after  #1802 to remove duplicated custom logic for Pods ownership by RayClusters.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: pkg/controller/jobs/pod/pod_controller_test.go [hunk 1]
```diff
@@ -3594,14 +3594,6 @@ func TestIsPodOwnerManagedByQueue(t *testing.T) {
 			},
 			wantRes: false,
 		},
-		"ray.io/v1alpha1/RayCluster": {
```
Comment (alculquicondor) [inline pkg/controller/jobs/pod/pod_controller_test.go:3597]: Can we keep the test case?

Comment (tenzen-y) [reply pkg/controller/jobs/pod/pod_controller_test.go:3597]: +1

**Review by tenzen-y — COMMENTED** [review]
LGTM

File: pkg/controller/jobs/pod/pod_controller_test.go (hunk 1 above)
Comment (astefanutti) [reply pkg/controller/jobs/pod/pod_controller_test.go:3597]: Yes, you're right. Updated. PTAL.

**Review by alculquicondor — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**